lib/TPP/Dialect/Xsmm/XsmmUtils.cpp

```diff
@@ -317,8 +318,10 @@ FailureOr<FusedMatch> getFusedBrgemmSequenceFromProducer(Operation *op) {
   // We don't know how to fuse more than two tail ops after the BRGEMM
   if (chain.size() > 3)
     return failure();
-  // List is in reverse order, put the brgemm at the top
-  std::reverse(chain.begin(), chain.end());
+  if (!isa<xsmm::BrgemmOp>(chain[0]))
```

Comment:
I don't understand this change. In which case the BRGEMM ends up as the first op from the loop above? Is this now coming in program order?

Reply:
Yes, this was in the integration test where the uses were traversed with brgemm first.